server/src/main/java/org/elasticsearch/index/mapper/MapperService.java

```diff
@@ -121,7 +121,7 @@ public MapperService(IndexSettings indexSettings, IndexAnalyzers indexAnalyzers,
         super(indexSettings);
         this.indexVersionCreated = indexSettings.getIndexVersionCreated();
         this.indexAnalyzers = indexAnalyzers;
-        this.indexAnalyzer = new MapperAnalyzerWrapper(indexAnalyzers.getDefaultIndexAnalyzer(), MappedFieldType::indexAnalyzer);
+        this.indexAnalyzer = new MapperAnalyzerWrapper();
```

Comment:
Something that bugs me, not directly related to your change, but that we may want to address: MapperService exposes indexAnalyzer, MappingLookup too. But they are two fundamentally different things. I have issues remembering which is what. Could we name them differently. Also as I am going to need to expose them both in the QueryShardContext. I am calling the new one getFieldNameAnalyzer, but you'll have better ideas.

Reply:
Yeah, I think this should be removed from MappingLookup entirely, and just live in MapperService.  I'll see if I can incorporate that here as well.

Reply:
This turned out not to be straightforward, so let's leave it for a separate issue

Reply:
++